src/ConvNetSharp.Core/Ops.cs

```diff
@@ -43,25 +47,10 @@ static Ops()
             Cast = Expression.Lambda<Func<int, T>>(castBody, intOperand).Compile();
 
             var logMethod = typeof(Math).GetRuntimeMethod("Log", new[] {typeof(T)});
-            
-            if (typeof(T) == typeof(double))
-            {
-                Log =
-                    Expression.Lambda<Func<T, T>>(
-                        Expression.Convert(
-                            Expression.Call(null, logMethod, Expression.Convert(firstOperand, typeof(double))),
-                            typeof(T)), firstOperand).Compile();
-                One = (T)(ValueType)1.0;
-            }
-            else if (typeof(T) == typeof(float))
-            {
-                Log =
-                    Expression.Lambda<Func<T, T>>(
-                        Expression.Convert(
-                            Expression.Call(null, logMethod, Expression.Convert(firstOperand, typeof(double))),
-                            typeof(T)), firstOperand).Compile();
-                One = (T)(ValueType)1.0f;
-            }
+            Log = Expression.Lambda<Func<T, T>>(
+                Expression.Convert(
+                    Expression.Call(null, logMethod, Expression.Convert(firstOperand, typeof(T))),
+                    typeof(T)), firstOperand).Compile();
```

Comment:
`Math.Log` method only exists for double type. That why I had to convert to double first when T is a float. I believe that it will raise an exception in single precision mode in the current state.